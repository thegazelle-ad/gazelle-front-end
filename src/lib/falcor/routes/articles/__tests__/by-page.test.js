import { articlesByPageRoutes } from '../by-page';

jest.enableAutomock();
jest.unmock('../by-page');
jest.unmock('lodash');
jest.unmock('falcor');

describe('articlesByPage falcor routes', () => {
  // More to make sure we didn't miss any
  it('has only 1 route', () => {
    expect(articlesByPageRoutes.length).toBe(1);
  });

  it('is the right route we are testing', () => {
    expect(articlesByPageRoutes[0].route).toBe(
      "articles['byPage'][{integers:pageLengths}][{integers:pageIndices}][{integers:indicesOnPage}]",
    );
  });

  it('returns correct path values', async () => {
    const pageLength = 5;
    const pageIndex = 2;
    const pathValues = await articlesByPageRoutes[0].get({
      pageLengths: [pageLength],
      pageIndices: [pageIndex],
      indicesOnPage: [0, 1, 2, 3, 4],
    });
    expect(pathValues).toMatchSnapshot();
  });

  it('throws on several pages requested', () =>
    expect(
      articlesByPageRoutes[0].get({
        pageLengths: [1],
        pageIndices: [1, 2],
        indicesOnPage: [0],
      }),
    ).rejects.toThrow());

  it('throws on negative page index requested', () =>
    expect(
      articlesByPageRoutes[0].get({
        pageLengths: [1],
        pageIndices: [-1],
        indicesOnPage: [0],
      }),
    ).rejects.toThrow());

  it('throws on negative pageLength requested', () =>
    expect(
      articlesByPageRoutes[0].get({
        pageLengths: [-1],
        pageIndices: [2],
        indicesOnPage: [0],
      }),
    ).rejects.toThrow());

  it('throws on several page lengths requested', () =>
    expect(
      articlesByPageRoutes[0].get({
        pageLengths: [1, 2],
        pageIndices: [2],
        indicesOnPage: [0],
      }),
    ).rejects.toThrow());

  it('throws when not all elements of a page were requested', () =>
    expect(
      articlesByPageRoutes[0].get({
        pageLengths: [2],
        pageIndices: [2],
        indicesOnPage: [0],
      }),
    ).rejects.toThrow());

  it('throws when out of range element on page is requested', () =>
    expect(
      articlesByPageRoutes[0].get({
        pageLengths: [2],
        pageIndices: [2],
        indicesOnPage: [0, 2],
      }),
    ).rejects.toThrow());

  it('throws when negative index on page is requested', () =>
    expect(
      articlesByPageRoutes[0].get({
        pageLengths: [2],
        pageIndices: [2],
        indicesOnPage: [-1, 0],
      }),
    ).rejects.toThrow());

  it('throws if duplicate indices on page are requested', () =>
    expect(
      articlesByPageRoutes[0].get({
        pageLengths: [2],
        pageIndices: [2],
        indicesOnPage: [0, 0],
      }),
    ).rejects.toThrow());
});
