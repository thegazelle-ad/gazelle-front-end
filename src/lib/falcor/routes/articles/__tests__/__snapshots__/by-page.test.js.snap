// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`articlesByPage falcor routes returns correct path values 1`] = `
Array [
  Object {
    "path": Array [
      "articles",
      "byPage",
      5,
      2,
      0,
    ],
    "value": Object {
      "$type": "ref",
      "value": Array [
        "articles",
        "bySlug",
        "slug-10",
      ],
    },
  },
  Object {
    "path": Array [
      "articles",
      "byPage",
      5,
      2,
      1,
    ],
    "value": Object {
      "$type": "ref",
      "value": Array [
        "articles",
        "bySlug",
        "slug-11",
      ],
    },
  },
  Object {
    "path": Array [
      "articles",
      "byPage",
      5,
      2,
      2,
    ],
    "value": Object {
      "$type": "ref",
      "value": Array [
        "articles",
        "bySlug",
        "slug-12",
      ],
    },
  },
  Object {
    "path": Array [
      "articles",
      "byPage",
      5,
      2,
      3,
    ],
    "value": Object {
      "$type": "ref",
      "value": Array [
        "articles",
        "bySlug",
        "slug-13",
      ],
    },
  },
  Object {
    "path": Array [
      "articles",
      "byPage",
      5,
      2,
      4,
    ],
    "value": Object {
      "$type": "ref",
      "value": Array [
        "articles",
        "bySlug",
        "slug-14",
      ],
    },
  },
]
`;
